GeoLib/AABB.h

```diff
@@ -108,9 +138,9 @@ class AABB
 	template <typename T>
 	bool containsPoint(T const & pnt) const
 	{
-		if (pnt[0] < _min_pnt[0] || _max_pnt[0] < pnt[0]) return false;
-		if (pnt[1] < _min_pnt[1] || _max_pnt[1] < pnt[1]) return false;
-		if (pnt[2] < _min_pnt[2] || _max_pnt[2] < pnt[2]) return false;
+		if (pnt[0] < _min_pnt[0] || _max_pnt[0] <= pnt[0]) return false;
+		if (pnt[1] < _min_pnt[1] || _max_pnt[1] <= pnt[1]) return false;
+		if (pnt[2] < _min_pnt[2] || _max_pnt[2] <= pnt[2]) return false;
```

Comment:
The maximal point is internally updated such that point located at the right boundaries (back, right or top surface) will fit into the aabb.

The problem with the previous version was that when the bounding box was created from a single point (min point == max point) the point was not in the bounding box itself.